src/js/components/header/customDomainModal.tsx

```diff
@@ -84,14 +89,30 @@ class CustomDomainModal extends React.Component<Props, { url: string }> {
               data-testid="custom-domain"
             >
               https://
-              {this.state.url.trim() ? (
-                extractCustomDomain(this.state.url.trim())
-              ) : (
-                <em>domain</em>
-              )}
+              {this.state.customDomain || <em>domain</em>}
               .my.salesforce.com
             </div>
           </Input>
+
+          <input
+            type="hidden"
+            name="csrfmiddlewaretoken"
+            value={cookies.get('csrftoken')}
+          />
+          <input
+            type="hidden"
+            name="next"
+            value={addUrlParams(
+              window.location.href,
+              this.props.redirectParams,
+            )}
+            data-testid="custom-login-next"
+          />
+          <input
+            type="hidden"
+            name="custom_domain"
+            value={this.state.customDomain}
+          />
```

Comment:
Processing the `customDomain` via POST depends on https://github.com/SFDO-Tooling/sfdo-template-helpers/pull/44